src/modules/core/index.ts

```diff
@@ -0,0 +1,54 @@
+import { Module } from 'vuex'
+import { RootState } from '@/store'
+import { geolocationObservable, checkGeolocationPermission } from '@/utils'
+
+interface State {
+  geolocation: Coordinates | null
+  geolocationPermission: string
+}
+
+const module: Module<State, RootState> = {
+  state: {
+    geolocation: null,
+    geolocationPermission: 'prompt',
+  },
+  actions: {
+    geolocate({ commit, dispatch }) {
+      return geolocationObservable.takeFirst().catch(e => {
+        dispatch('message', 'No se pudo acceder a la geolocalizacion')
+        console.error(e)
+        return Promise.reject(e)
```

Comment:
Ahora la promise puede ser rejected si falla la geolocalizacion. Que opinas de este catch chaining? Quizas esta action no haria falta y se podria usar directamente el geolocationObservable, pero capaz esta bueno tener este "middleware" que hace dispatch del mensaje de error y pasar el catch hacia el siguiente handler para manejar el error segun le convenga al que llame a esta action

Reply:
Si, me parece que esta mejor asi, capaz en este caso no haria falta el console.error(), porque ya estas volviendo a hacer reject?

Reply:
si no lo cacheas te tira un error en consola, `Uncaught (in promise) PositionError {code: 1, message: "User denied Geolocation"}` pero muestra el mensaje de error por el middleware y lo demas sigue andando